cabal-testsuite/PackageTests/OfflineFlag/offlineFlag.out

```diff
@@ -0,0 +1,25 @@
+# cabal v2-update
+Downloading the latest package list from test-local-repo
+# cabal v2-build
+Resolving dependencies...
+Build profile: -w ghc-<GHCVER> -O1
+In order, the following will be built:
+ - remote-0.1.0.0 (lib) (requires build)
+ - current-0.1.0.0 (exe:current) (first run)
+cabal: --offline was specified, could not download package remote version 0.1.0.0
+CallStack (from HasCallStack):
```

Comment:
Why do we get an exception? This shouldn't be the case, I think. If there's a remote dependency that is not in the cache and we received `--offline`, we need to gracefully shutdown printing a helpful message. The message is already up (see the line above). But the shutdown shouldn't bring up a stack trace!

Reply:
As stated above, I changed the build failure type to take a custom string and not produce an exception.